library/ziti.c

```diff
@@ -416,6 +421,26 @@ void ziti_set_fully_authenticated(ziti_context ztx, const char *session_token) {
         ziti_ctrl_create_api_certificate(ztx_get_controller(ztx), ztx->sessionCsr, on_create_cert, ztx);
     }
 
+    if (ztx->opts.enable_cert_extension && ztx->id_creds.cert) {
+        struct tm exp;
+
+        ztx->id_creds.cert->get_expiration(ztx->id_creds.cert, &exp);
+        time_t now = time(0);
+        time_t exptime = mktime(&exp);
+
+        bool renew = exptime - now < TEN_DAYS;
```

Comment:
why not make TEN_DAYS a config option on `ziti_options_s` and let the tunneler/client control this?

Reply:
is this better?